src/tlo/methods/enhanced_lifestyle.py

```diff
@@ -825,533 +428,1495 @@ def on_birth(self, mother_id, child_id):
+                                       base_prob,
+                                       Predictor().when('(age_years >= 15) & (sex == "M")',
+                                                        self.params['init_p_ex_alc_m'])
+                                       .when('(age_years >= 15) & (sex == "F")',
+                                             self.params['init_p_ex_alc_f']),
+                                       )
+        # return excessive alcohol linear model
+        return excessive_alc_lm
+
+    def marital_status_linear_model(self) -> LinearModel:
+        """A function to create linear model for individual's marital status. Here, We are using a custom
+        linear model and we are assigning individual's marital status based on their age group.In this module,
+        marital status is in three categories;
+                1.  Never Married
+                2.  Currently Married
+                3   Divorced or Widowed
+        """
 
-        # -------------------- EDUCATION -----------------------------------------------------------
+        def init_marital_status(self, df, rng=None, **externals) -> pd.Series:
+            """ a function to assign marital status to individuals of different age groups
+
+            :param self: a reference to cutom linear model
+            :param df:  population dataframe
+            :param  rng:
+            :param  externals a dict containing any other variables passed on to this function """
+            # get access to Lifestyle module parameters
+            p = self.parameters
+
+            # create a new series to hold individual's marital status
+            mar_stat = pd.Series(data=1, index=df.index, dtype=int)
+
+            # select individuals of different age category
+            age_15_19 = df.index[df.age_years.between(15, 19) & df.is_alive]
+            age_20_29 = df.index[df.age_years.between(20, 29) & df.is_alive]
+            age_30_39 = df.index[df.age_years.between(30, 39) & df.is_alive]
+            age_40_49 = df.index[df.age_years.between(40, 49) & df.is_alive]
+            age_50_59 = df.index[df.age_years.between(50, 59) & df.is_alive]
+            age_gte60 = df.index[(df.age_years >= 60) & df.is_alive]
+
+            mar_stat.loc[age_15_19] = rng.choice([1, 2, 3], size=len(age_15_19), p=p['init_dist_mar_stat_age1520'])
+            mar_stat.loc[age_20_29] = rng.choice([1, 2, 3], size=len(age_20_29), p=p['init_dist_mar_stat_age2030'])
+            mar_stat.loc[age_30_39] = rng.choice([1, 2, 3], size=len(age_30_39), p=p['init_dist_mar_stat_age3040'])
+            mar_stat.loc[age_40_49] = rng.choice([1, 2, 3], size=len(age_40_49), p=p['init_dist_mar_stat_age4050'])
+            mar_stat.loc[age_50_59] = rng.choice([1, 2, 3], size=len(age_50_59), p=p['init_dist_mar_stat_age5060'])
+            mar_stat.loc[age_gte60] = rng.choice([1, 2, 3], size=len(age_gte60), p=p['init_dist_mar_stat_agege60'])
```

Comment:
```suggestion
            li_mar_stat_dtype = df.li_mar_stat.dtype
            mar_stat = pd.Series(data=1, index=df.index, dtype=li_mar_stat_dtype )

            # select individuals of different age category
            age_ranges = [(15, 20), (20, 30), (30, 40), (40, 50), (50, 60), (60, np.inf)]
            for lower_age, upper_age in age_ranges:
                subpopulation = df.index[
                    df.age_years.between(lower_age, upper_age, inclusive="left")
                    & df.is_alive
                ]
                parameters_key = (
                    f"init_dist_mar_stat_age{lower_age}{upper_age}"
                    if upper_age != np.inf else  
                    f"init_dist_mar_stat_agege{upperage}"
                )
                mar_stat[subpopulation] = rng.choice(
                    li_mar_stat_dtype.categories,
                    p=p[parameters_key],
                    size=len(subpopulation)
                )
```
I think above should have the same result as the original implementation while avoiding the repetition and ensuring the categories for `li_mar_stat` are not hard coded in multiple places.

Reply:
great, here  
 ```
 else  
    f"init_dist_mar_stat_agege{upperage}"
```
I think the variable should be `lower_age` instead of `upperage`  as we don't want to get `np.inf`?